Sources/Apollo/GraphQLQueryWatcher.swift

```diff
@@ -39,26 +40,32 @@ public final class GraphQLQueryWatcher<Query: GraphQLQuery>: Cancellable, Apollo
   private let callbackQueue: DispatchQueue = .main
 
   func fetch(cachePolicy: CachePolicy) {
-    // Cancel anything already in flight before starting a new fetch
-    fetching?.cancel()
-    fetching = client?.fetch(query: query, cachePolicy: cachePolicy, contextIdentifier: self.contextIdentifier, queue: callbackQueue) { [weak self] result in
-      guard let self = self else { return }
+    lock.withLock {
```

Comment:
It isn't. The fetch is asynchronous, so it only locks the initial call, not the completion. The lock needs to be around the call because the assignment to `self.fetching` could otherwise lead to race conditions.

Reply:
why not use `Atomic` on `fetching` instead? That's wayyyy more lightweight

Reply:
I didn't think that would work here, because you want both the `fetching?.cancel()` and the `fetching =` to be locked together. I now see there is a `mutate` method, so that might be an option, but that means the scope would have to be the same as the current `withLock`, so that doesn't seem more lightweight.

Reply:
I'm less concerned about lightweight than I am about making the purpose of the lock clearer - with this the lock is just slapped on everything so it's harder to tell what it's targeting.

Reply:
Although I like the idea of an `Atomic` wrapper (especially if we can make it a real property wrapper), it does have some problems of its own that we should be aware of. The current implementation isn't truly thread-safe for example. Any operation that relies on both getting and setting the value isn't atomic. So even something simple like `flag.value.toggle()` isn't safe, and using collection types (`list.value.append(element)`) is even worse. So at a minimum we should remove the setter and rely on `mutate` for any changes.

Reply:
Talked this through and it makes sense - I'm going to leave the convenience getter and kill the convenience setter to prevent misuse in a separate PR.